internal/pkg/deploy/cloudformation/app.go

```diff
@@ -312,6 +312,51 @@ func (cf CloudFormation) RemoveJobFromApp(app *config.Application, jobName strin
 	return nil
 }
 
+// RemoveEnvFromAppOpts contains the parameters to call RemoveEnvFromApp.
+type RemoveEnvFromAppOpts struct {
+	App                 *config.Application
+	EnvName             string
+	EnvAccountID        string
+	EnvRegion           string
+	DeleteStackInstance bool // If this is the last environment in a region, we can delete it.
+}
+
+func (cf CloudFormation) RemoveEnvFromApp(opts *RemoveEnvFromAppOpts) error {
```

Comment:
What do you think of renaming this function to `RemoveLastEnvInAccountRegionFromApp` (crappy name but...) then we can remove the `DeleteStackInstance` from `opts` 💭

Reply:
I think it's possible to have the situation where we don't need to delete a stackset instance but we do need to redeploy the app and other stackset instances to remove an account from IAM roles and DNS delegation. I don't love having a function call which might be a no-op but everything should be idempotent.